mqtt/mqtt3/src/proto/packet.rs

```diff
@@ -137,6 +138,20 @@ pub struct Connect {
     pub protocol_level: u8,
 }
 
+impl Default for Connect {
```

Comment:
I don't think it is a good idea to add a default for proto::Connect only for testing purposes. Semantically connect packet suppose to have data that identifies a client.
Add the factory method in tests instead if you need to

Reply:
Agreed, will add the factory method.